### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by mchamsteryt2** [Rem0o/FanControl.Releases#631] [Q&A]
What Are The Needed Parts For The HWI Plugin

**Comment by Fr0stX76** [comment]
First, download the file from here https://ci.appveyor.com/project/Rem0o/fancontrol-hwinfo/branch/master/artifacts

While fancontrol is closed, unzip the files from the zip in the plugin folder of Fancontrol, if needed, unblock the dll (see [here](http://www.itlec.com/2016/05/systemnotsupportedexception-attempt-was.html)), start fancontrol, activate the plugin in the sensor menu (top right 3 dots) Then refresh sensors... 

Voila

**Reply by mchamsteryt2** [reply]
Do I Need Specific Sensors For HWI

**Reply by Fr0stX76** [reply]
No. any temp sensor not hidden in hwinfo should show up in fancontrol

**Reply by ChaplainTIG** [reply]
I'm not seeing any option to activate the plugin. Feel like I'm missing something simple. Any ideas?
![Screenshot 2022-04-23 185426](https://user-images.githubusercontent.com/33890682/164952447-92474d52-fd73-423d-b676-5e0972ecdaa8.jpg)

**Reply by Fr0stX76** [reply]
Enabling shared memory in HWinfo?

**Reply by ChaplainTIG** [reply]
Thank you :) Did that and restarted, but still no change.
Looked at the log:
4/23/2022 10:06:35 PM: System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
4/23/2022 10:06:35 PM: Failed to load plugin FanControl.Plugins.dll.
4/23/2022 10:06:35 PM: System.IO.FileLoadException: Could not load file or assembly 'file:///C:\Program Files\FanControl\Plugins\FanControl.Plugins.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
File name: 'file:///C:\Program Files\FanControl\Plugins\FanControl.Plugins.dll' ---> System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at FanControl.Domain.PluginService.LoadAssembly(FileInfo dll)


4/23/2022 10:06:35 PM: System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.

**Reply by Rem0o** [reply]
Unlock the hwinfo dll . Right click properties

**Reply by ghost** [reply]
Are you using the x64 version of HWInfo?  If so, the x64 version now limits how long shared memory is active - unless you pay :(

Also, the shared memory file is only available when the sensors are running.  Just running HWInfo is not enough, you can tick the box "Show Sensors on Startup" to make sure this always runs.

If you have already done that then you might need to do what I had to, edit the application .config file: "FanControl.exe.config" - it's an XML file used by the .Net Framework, simply add <loadFromRemoteSources  enabled="true"/> to the <runtime> node - the order of items in there matters not, but don't delete anything else or FanControl will not load.

**Reply by ChaplainTIG** [reply]
Thank you @Rem0o that did it :D

**Reply by ChaplainTIG** [reply]
And thank you too @iJacks1980 
When I checked, the "Show Sensors on Startup" was already selected, and I happily paid up to get access to my temp sensors in Fan Control.
Greatly appreciate your help.

**Reply by ChaplainTIG** [reply]
One final challenge.
I've got a fan plugged into the #7 slot that pulls air from the a/c system when I'm rendering. I can see "System 7" fan on HWI and in the "Speeds" section of Fan Control, but it's not showing up in the "Controls" area. Any ideas? And thank you very much for all this hand holding getting this God-send of a program all setup for me. You absolutely have another supporter here!! Paypal incoming ;)
